crates/chia-consensus/fuzz/fuzz_targets/solution-generator.rs

```diff
@@ -0,0 +1,42 @@
+#![no_main]
+use chia_consensus::gen::solution_generator::{calculate_generator_length, solution_generator};
+use chia_protocol::{Coin, CoinSpend};
+use chia_traits::Streamable;
+use clvmr::{
+    serde::{node_from_bytes_backrefs, node_to_bytes},
+    Allocator,
+};
+use libfuzzer_sys::fuzz_target;
+use std::io::Cursor;
+
+fuzz_target!(|data: &[u8]| {
+    let mut spends = Vec::<CoinSpend>::new();
+    let mut generator_input = Vec::<(Coin, Vec<u8>, Vec<u8>)>::new();
+    let mut data = Cursor::new(data);
+    let mut discrepancy: i64 = 0;
+    let mut a = Allocator::new();
+    while let Ok(spend) = CoinSpend::parse::<false>(&mut data) {
+        spends.push(spend.clone());
+        generator_input.push((
+            spend.coin,
+            spend.puzzle_reveal.to_vec(),
+            spend.solution.to_vec(),
+        ));
+        // Check if puzzle or solution are atoms which can represented in a smaller form
```

Comment:
this comment is technically outdated now. It doesn't matter whether the atom is at the top level or buried deep down a tree. it still introduces the same discrepancy